src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1658,8 +1668,18 @@ private void PopStackFrame(EvalStackFrame frame, bool mergeNestedRegions = true)
                     Debug.Assert(toMerge.Locals.IsEmpty);
 
                     _currentRegion.AddCaptureIds(toMerge.CaptureIds);
-                    _currentRegion.ExtendToInclude(toMerge.LastBlock);
-                    MergeSubRegionAndFree(toMerge, _blocks, _regionMap);
+                    // This region can be empty in certain error scenarios, such as `new T {}`, where T does not
+                    // have a class constraint. There are no arguments or initializers, so nothing will have
+                    // been put into the region at this point
+                    if (toMerge.LastBlock is null)
+                    {
+                        Debug.Assert(toMerge.FirstBlock is null);
+                    }
+                    else
+                    {
+                        _currentRegion.ExtendToInclude(toMerge.LastBlock);
+                    }
+                    MergeSubRegionAndFree(toMerge, _blocks, _regionMap, canHaveEmptyRegion: true);
```

Comment:
>true [](start = 92, length = 4)

Consider setting `canHaveEmptyRegion: toMerge.LastBlock is null`, to catch the (impossible?) case where `LastBlock != null` and `FirstBlock == null`.

Reply:
I'll change the `if`. One but not both were `null`, many assumptions of the builder would be violated. I'd expect us to crash with an NPE in that case.

---
In reply to: [239627311](https://github.com/dotnet/roslyn/pull/31592#discussion_r239627311) [](ancestors = 239627311)